packages/edit-site/src/components/global-styles/font-size.js

```diff
@@ -0,0 +1,239 @@
+import ScreenHeader from './header';
+
+function FontSizePreview( { fontSize } ) {
+	const [ font ] = useGlobalStyle( 'typography' );
+	return (
+		<div
+			className="edit-site-typography-preview"
+			style={ {
+				fontSize: fontSize.size,
+				fontFamily: font?.fontFamily ?? 'serif',
+			} }
+		>
+			{ __( 'Aa' ) }
+		</div>
+	);
+}
+
+function FontSize() {
+	const {
+		params: { slug },
+		goBack,
+	} = useNavigator();
+	const [ fontSizes, setFontSizes ] = useGlobalSetting(
+		'typography.fontSizes'
+	);
+	// Get the font sizes from the theme or use the default ones.
+	const sizes = fontSizes.theme ?? fontSizes.default;
+
+	// Get the font size by slug.
+	const fontSize = sizes.find( ( size ) => size.slug === slug );
+
+	const hasUnits = typeof fontSize.size === 'string';
+
+	// Whether fluid is true or an object, set it to true, otherwise false.
+	const isFluid = !! fontSize.fluid ?? false;
+
+	// Whether custom fluid values are used.
+	const isCustomFluid = typeof fontSize.fluid === 'object';
+
+	// Initialize the ref with the prop value on the first render
+	const initialFontSizeRef = useRef( fontSize.size );
+
+	// Memoized initial value of fontSize
+	const initialFontSize = initialFontSizeRef.current;
+
+	const handleNameChange = ( value ) => {
+		updateFontSize( 'name', value );
+	};
+
+	const handleFontSizeChange = ( value ) => {
+		// If the user is resetting the value, use the initial value.
+		const newValue = value ?? initialFontSize;
+		updateFontSize( 'size', newValue );
+	};
+
+	const handleFluidChange = ( value ) => {
+		updateFontSize( 'fluid', value );
+	};
+
+	const handleCustomFluidValues = ( value ) => {
+		if ( value ) {
+			// If custom values are use init the values with the current ones.
+			updateFontSize( 'fluid', {
+				min: fontSize.size,
+				max: fontSize.size,
+			} );
+		} else {
+			// If custom fluid values are disabled, set fluid to true.
+			updateFontSize( 'fluid', true );
+		}
+	};
+
+	const handleMinChange = ( value ) => {
+		// If the user is resetting the value, use the initial value.
+		const newValue = value ?? fontSize.size;
+		updateFontSize( 'fluid', { ...fontSize.fluid, min: newValue } );
+	};
+
+	const handleMaxChange = ( value ) => {
+		// If the user is resetting the value, use the initial value.
+		const newValue = value ?? fontSize.size;
+		updateFontSize( 'fluid', { ...fontSize.fluid, max: newValue } );
+	};
+
+	const updateFontSize = ( key, value ) => {
+		const newFontSizes = sizes.map( ( size ) => {
+			if ( size.slug === slug ) {
+				return { ...size, [ key ]: value }; // Create a new object with updated key
+			}
+			return size;
+		} );
+
+		setFontSizes( {
+			...fontSizes,
+			theme: newFontSizes,
+		} );
+	};
+
+	const handleRemoveFontSize = () => {
+		// Navigate to the font sizes list.
+		goBack();
+
+		const newFontSizes = sizes.filter( ( size ) => size.slug !== slug );
+		setFontSizes( {
+			...fontSizes,
+			theme: newFontSizes,
```

Comment:
Similar to this comment.

Reply:
Let's continue discussing this in that thread https://github.com/WordPress/gutenberg/pull/62328#discussion_r1646887376. Having multiple threads for the same thing make things harder to read.